monkey/infection_monkey/ransomware/ransomware_payload.py

```diff
@@ -55,7 +56,7 @@ def _find_files(self) -> List[Path]:
             return []
 
         return select_production_safe_target_files(
-            Path(self._target_dir), self._valid_file_extensions_for_encryption
+            Path(os.path.expandvars(self._target_dir)), self._valid_file_extensions_for_encryption
```

Comment:
We should use `expand_path()` from `monkey_island/cc/server_utils/file_utils.py` It also expands "\~" to "$HOME". We should probably add a validator to the Linux target directory input to allow paths that begin with "~".